index.js

```diff
@@ -4,55 +4,75 @@ const Koa = require('koa');
 const Router = require('@koa/router');
 const bodyParser = require('koa-bodyparser');
 const axios = require('axios');
+const yaml = require('js-yaml');
+const fs = require('fs');
 
 const port = process.env.PORT || 5001;
-const hook = process.env.DISCORD_WEBHOOK;
+const configPath = "/etc/alertmanager-discord.yml";
 const hookRegExp = new RegExp('https://discord(?:app)?.com/api/webhooks/[0-9]{18}/[a-zA-Z0-9_-]+');
 const colors = {firing: 0xd50000, resolved: 0x00c853};
 const maxEmbedsLength = 10;
+const routes = {};
 
-async function handleIndex(ctx) {
-  if (ctx.request.body && Array.isArray(ctx.request.body.alerts)) {
-    const embeds = [];
-
-    ctx.request.body.alerts.forEach(alert => {
-      if (alert.annotations && (alert.annotations.summary || alert.annotations.description)) {
-        embeds.push({
-          title: alert.annotations.summary,
-          description: alert.annotations.description,
-          color: alert.status === 'resolved' ? colors.resolved : colors.firing,
-        });
-      }
-    });
+async function handleHook(ctx) {
+  let hook = routes[ctx.params.slug];
 
-    if (!embeds.length) {
-      ctx.status = 400;
-      console.warn('No data to write to embeds');
-      return;
-    }
+  if (hook === undefined) {
+    ctx.status = 404;
+    console.warn(`Slug "${ctx.params.slug}" was not found in routes`);
+    return;
+  }
 
-    let chunk = [];
-    while ((chunk = embeds.splice(0, maxEmbedsLength)) && chunk.length) {
-      await axios.post(
-        hook,
-        {embeds: chunk},
-      ).then(() => {
-        ctx.status = 200;
-        console.log(chunk.length + ' embeds sent');
-      }).catch(err => {
-        ctx.status = 500;
-        console.error(err);
-        return;
+  if (ctx.request.body === undefined || !Array.isArray(ctx.request.body.alerts)) {
+    ctx.status = 400;
+    console.error('Unexpected request from Alertmanager:', ctx.request.body);
+    return;
+  }
+
+  const embeds = [];
+
+  ctx.request.body.alerts.forEach(alert => {
+    if (alert.annotations && (alert.annotations.summary || alert.annotations.description)) {
```

Comment:
Why OR in `alert.annotations.summary || alert.annotations.description` if we use both at the end?

Reply:
If for some reason we omit any one of these fields it is not a problem and empty one will be just ignored. But with both summary and description empty fields notification has no sense.